google/cloud/spanner_v1/pool.py

```diff
@@ -138,6 +157,115 @@ def session(self, **kwargs):
+        ):
+            AbstractSessionPool._cleanup_task_ongoing_event.set()
+            if self._database.logging_enabled:
+                self._database.logger.info("95% of the session pool is exhausted")
+            background = threading.Thread(
+                target=self.deleteLongRunningTransactions,
+                args=(
+                    DELETE_LONG_RUNNING_TRANSACTION_FREQUENCY_SEC,
+                    DELETE_LONG_RUNNING_TRANSACTION_THRESHOLD_SEC,
+                ),
+                daemon=True,
+                name="recycle-sessions",
+            )
+            background.start()
+        else:
+            AbstractSessionPool._cleanup_task_ongoing_event.set()
+
+    def stopCleaningLongRunningSessions(self):
+        """Stops background task for recycling session."""
+        AbstractSessionPool._cleanup_task_ongoing_event.clear()
+
+    def deleteLongRunningTransactions(
+        self, longRunningTransactionFrequency_sec, longRunningTransactionThreshold_sec
+    ):
+        """Recycles sessions with long-running transactions
+        :param longRunningTransactionFrequency_sec: Interval for running background task in seconds.
+        :param longRunningTransactionThreshold_sec: Timeout for recycling sessions in seconds.
+        """
+        long_running_transaction_timer = time.time()
+        transactions_closed = 0
+        while AbstractSessionPool._cleanup_task_ongoing_event.is_set():
+            AbstractSessionPool._cleanup_task_ongoing = True
+            is_timeout_reached = (
+                time.time() - long_running_transaction_timer
+                >= longRunningTransactionThreshold_sec
+                + longRunningTransactionFrequency_sec
+            )
+
+            if is_timeout_reached and transactions_closed == 0:
+                break
+            iteration_start = time.time()
+
+            # Retrieve a list of sessions to delete.
+            sessions_to_delete = [
+                session
+                for session in self._borrowed_sessions
+                if (datetime.datetime.utcnow() - session.checkout_time)
+                > datetime.timedelta(seconds=longRunningTransactionThreshold_sec)
+                and not session.long_running
+            ]
+
+            for session in sessions_to_delete:
+                transactions_closed += int(self._recycle_session(session))
+
+            # Calculate and sleep for the time remaining until the next iteration based on the interval.
+            iteration_elapsed = time.time() - iteration_start
+            remaining_time = longRunningTransactionFrequency_sec - iteration_elapsed
+            if remaining_time > 0:
+                time.sleep(remaining_time)
+
+        AbstractSessionPool._cleanup_task_ongoing = False
+        AbstractSessionPool._cleanup_task_ongoing_event.clear()
+
+    def _recycle_session(self, session):
+        """Helper method to recycle session for background task.
```

Comment:
1. We can avoid "background task" in method documentation.
2. Instead of "recycle_session" which is again a new terminology can we keep it consistent? We are already using close_long_running_sessions / close_long_running_transactions.

Reply:
Done